Build error: EGL/egl.h not found (buildroot)

### Tested versions

4.x as of 1bffd6c73b44b85e5889f54e14b2193940cf5bb1

### System information

Buildroot version: godot-2023.08.x-3

### Issue description

```
...
Generating modules/modules_enabled.gen.h ...
Compiling thirdparty/openxr/src/common/filesystem_utils.cpp ...
In file included from thirdparty/openxr/src/common/platform_utils.hpp:12,
                 from thirdparty/openxr/src/common/filesystem_utils.cpp:13:
thirdparty/openxr/src/common/xr_dependencies.h:80:10: fatal error: EGL/egl.h: No such file or directory
   80 | #include <EGL/egl.h>
```

`libegl1-mesa-dev` is installed on system path just in case.

### Steps to reproduce

Build from source `master` branch.

### Minimal reproduction project (MRP)

-

--- a/.github/workflows/linux_builds.yml
+++ b/.github/workflows/linux_builds.yml
@@ -106,7 +106,7 @@ jobs:
       # TODO: Figure out somehow how to embed this one.
       - name: wayland-scanner dependency
         run: |
-          sudo apt-get install libwayland-bin libegl-dev
+          sudo apt-get install libwayland-bin
 
       - name: Free disk space on runner
         run: |

--- a/thirdparty/openxr/patches/use-egl-from-glad.diff
+++ b/thirdparty/openxr/patches/use-egl-from-glad.diff
@@ -1,0 +1,27 @@
+diff -Nur openxr-orig/src/common/xr_dependencies.h openxr/src/common/xr_dependencies.h
+--- openxr-orig/src/common/xr_dependencies.h	2024-11-04 10:38:11.940682727 -0600
++++ openxr/src/common/xr_dependencies.h	2024-11-04 10:38:46.351415476 -0600
+@@ -65,7 +65,11 @@
+ #endif  // XR_USE_GRAPHICS_API_OPENGL
+ 
+ #ifdef XR_USE_GRAPHICS_API_OPENGL_ES
++#ifdef GLAD_ENABLED
++#include "thirdparty/glad/glad/egl.h"
++#else
+ #include <EGL/egl.h>
++#endif
+ #endif  // XR_USE_GRAPHICS_API_OPENGL_ES
+ 
+ #ifdef XR_USE_GRAPHICS_API_VULKAN
+@@ -77,7 +81,11 @@
+ #endif  // XR_USE_PLATFORM_WAYLAND
+ 
+ #ifdef XR_USE_PLATFORM_EGL
++#ifdef GLAD_ENABLED
++#include "thirdparty/glad/glad/egl.h"
++#else
+ #include <EGL/egl.h>
++#endif
+ #endif  // XR_USE_PLATFORM_EGL
+ 
+ #if defined(XR_USE_PLATFORM_XLIB) || defined(XR_USE_PLATFORM_XCB)

--- a/thirdparty/openxr/src/common/xr_dependencies.h
+++ b/thirdparty/openxr/src/common/xr_dependencies.h
@@ -65,7 +65,11 @@
 #endif  // XR_USE_GRAPHICS_API_OPENGL
 
 #ifdef XR_USE_GRAPHICS_API_OPENGL_ES
+#ifdef GLAD_ENABLED
+#include "thirdparty/glad/glad/egl.h"
+#else
 #include <EGL/egl.h>
+#endif
 #endif  // XR_USE_GRAPHICS_API_OPENGL_ES
 
 #ifdef XR_USE_GRAPHICS_API_VULKAN
@@ -77,7 +81,11 @@
 #endif  // XR_USE_PLATFORM_WAYLAND
 
 #ifdef XR_USE_PLATFORM_EGL
+#ifdef GLAD_ENABLED
+#include "thirdparty/glad/glad/egl.h"
+#else
 #include <EGL/egl.h>
+#endif
 #endif  // XR_USE_PLATFORM_EGL
 
 #if defined(XR_USE_PLATFORM_XLIB) || defined(XR_USE_PLATFORM_XCB)

